tests/server/api/v1/test_datasets.py

```diff
@@ -749,6 +752,165 @@ def test_create_dataset_with_existent_name(client: TestClient, db: Session, admi
     assert db.query(Dataset).count() == 1
 
 
+def test_create_dataset_field(client: TestClient, db: Session, admin_auth_header: dict):
+    dataset = DatasetFactory.create()
+    field_json = {
+        "name": "name",
+        "title": "title",
+        "settings": {"type": "text"},
+    }
+
+    response = client.post(f"/api/v1/datasets/{dataset.id}/fields", headers=admin_auth_header, json=field_json)
+
+    assert response.status_code == 201
+    assert db.query(Field).count() == 1
+
+    response_body = response.json()
+    assert db.get(Field, UUID(response_body["id"]))
+    assert response_body == {
+        "id": str(UUID(response_body["id"])),
+        "name": "name",
+        "title": "title",
+        "required": False,
+        "settings": {"type": "text"},
+        "inserted_at": datetime.fromisoformat(response_body["inserted_at"]).isoformat(),
+        "updated_at": datetime.fromisoformat(response_body["updated_at"]).isoformat(),
+    }
+
+
+def test_create_dataset_field_without_authentication(client: TestClient, db: Session):
+    dataset = DatasetFactory.create()
+    field_json = {
+        "name": "name",
+        "title": "title",
+        "settings": {"type": "text"},
+    }
+
+    response = client.post(f"/api/v1/datasets/{dataset.id}/fields", json=field_json)
+
+    assert response.status_code == 401
+    assert db.query(Field).count() == 0
+
+
+def test_create_dataset_field_as_annotator(client: TestClient, db: Session):
+    annotator = AnnotatorFactory.create()
+    dataset = DatasetFactory.create()
+    field_json = {
+        "name": "name",
+        "title": "title",
+        "settings": {"type": "text"},
+    }
+
+    response = client.post(
+        f"/api/v1/datasets/{dataset.id}/fields",
+        headers={API_KEY_HEADER_NAME: annotator.api_key},
+        json=field_json,
+    )
+
+    assert response.status_code == 403
+    assert db.query(Field).count() == 0
+
+
+@pytest.mark.parametrize("invalid_name", ["", " ", "  ", "-", "--", "_", "__", "A", "AA", "invalid_nAmE"])
+def test_create_dataset_field_with_invalid_name(
+    client: TestClient, db: Session, admin_auth_header: dict, invalid_name: str
+):
+    dataset = DatasetFactory.create()
+    field_json = {
+        "name": invalid_name,
+        "title": "title",
+        "settings": {"type": "text"},
+    }
+
+    response = client.post(f"/api/v1/datasets/{dataset.id}/fields", headers=admin_auth_header, json=field_json)
+
+    assert response.status_code == 422
```

Comment:
If the response content contains a more descriptive error, maybe makes sense to add a check of the expected info.

Reply:
It's not custom, it's the default Pydantic error message. We can check it but we have not been checking it for the rest of fields.